src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceIntTest.kt

```diff
@@ -714,6 +723,211 @@ class OffenderResourceIntTest : ResourceTest() {
     assertThatJsonFileAndStatus(response, 200, "adjudications_by_finding_code.json")
   }
 
+  @Nested
+  @DisplayName("PUT /api/offenders/{offenderNo}/case-notes/{caseNoteId}")
+  inner class UpdateCaseNote {
+    private val caseNoteUpdate =
+      """ 
+        {
+          "type": "CHAP",
+          "subType": "FAMMAR",
+          "text" : "Hello this is a case note"
+        }
+      """
+
+    @Test
+    fun `returns 401 without an auth token`() {
+      webTestClient.put().uri("/api/offenders/A1234AP/case-notes/34")
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .body(BodyInserters.fromValue(caseNoteUpdate))
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `returns 403 if no user`() {
+      webTestClient.put().uri("/api/offenders/A1234AP/case-notes/34")
+        .headers(setClientAuthorisation(listOf()))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .body(BodyInserters.fromValue(caseNoteUpdate))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `Attempt to update case note for offender that is not part of any of logged on staff user's caseloads`() {
+      webTestClient.put().uri("/api/offenders/A1234AP/case-notes/34")
+        .headers(setAuthorisation("ITAG_USER", listOf("")))
+        .header("Content-Type", APPLICATION_JSON_VALUE)
+        .body(BodyInserters.fromValue(caseNoteUpdate))
+        .exchange()
+        .expectStatus().isNotFound
+        .expectBody().jsonPath("userMessage").isEqualTo("Resource with id [A1234AP] not found.")
+    }
+
+    @Test
```

Comment:
Pulled in these tests from the old update case note by bookingId (which has now been removed as it is unused)